domain_utils/domain_utils.py

```diff
@@ -217,18 +238,20 @@ def get_stripped_url(url, scheme=False, drop_non_http=False, use_netloc=True, ex
         Returns a url stripped to (scheme)?+(netloc|hostname)+(path)?.
         Returns empty string if appropriate.
     """
-
     url = _adapt_url_for_port_and_scheme(url, extractor)
 
-    purl = urlparse(url)
+    purl = urlparse(url, scheme=scheme_default)
     _scheme = purl.scheme
 
-    # Handle non http schemes
-    if _scheme not in ['http', 'https', '']:
-        if drop_non_http is True:
-            return ''
-        else:
+    # Will we parse
+    schemes_to_parse = [HTTP, HTTPS]
+    if parse_ws is True:
+        schemes_to_parse += [WS, WSS]
+    if _scheme not in schemes_to_parse:
+        if return_unparsed is True:
             return url
+        else:
```

Comment:
The `else:` can be dropped here

Reply:
This will cause 3 tests to fail. Are you sure?

Reply:
Sorry. I meant the `else:` can be implicit. Either you match the condition and return early, or you return empty. Just a style comment.

Reply:
Sorry, I'm not sure I'm understanding what you mean. Do you mean, you prefer
```python
    if _scheme not in schemes_to_parse:
        if return_unparsed is True:
            return url
        return ''
```

Reply:
Yes

Reply:
HAH! I literally was struggling to figure that out. I'll change it. I'm not sure how I feel about it, but that's a good reason to try it on for size :)

Reply:
I could have been clearer!

It's not super important here, since the `else` condition is only one line, but I think this pattern is much more readable when the else condition is longer.

Reply:
My gut reaction was "explicit is better than implicit" (zen of python #2). I see your point though. I will mull on it. This may even call for a twitter poll.